### DISCUSSION TITLE
`service_control`의 사용처

### DISCUSSION THREAD

**Discussion by msk** [aicers/roxy#145] [Q&A] [answered]
`service_control`로 제어할 필요가 있는 서비스 목록을 알 수 있을까요?

현재는 roxy만 쓸 수 있으면 `service_control`로 아무 서비스나 제어할 수 있으므로, 그보다는 보안 측면에서 각 서비스별로 제어할 수 있는 API를 각각 만들고 `service_control`은 public API에서 제외하는 편이 좋을 것 같은데요.

**Comment by syncpark** [comment, marked as answer]
서비스 목록입니다.
* REview, REconverge, Hog, Giganto, Piglet, Crusher, Tivan
* (r)syslog, ssh, ufw, ntp, postgres(docker 버전이 아닌 경우에만)

**Reply by msk** [reply]
`service_control` 대신 각 서비스별로 API를 따로 만드는 것에 대해서는 어떻게 생각하시나요?

**Reply by syncpark** [reply]
지금 Roxy에서 syslog, ssh, ntp는 내부적으로 service_control()을 사용하지 않고, 직접 systemctl crate를 사용하도록 되어있습니다.
그외에 Roxy의 service_control()을 호출하는 경우는 `REview, ..., Tivan`과 UFW를 관리할 때입니다.

남은 이슈:
1. REview, REconverge 들은 어떻게 하는게 좋을까요? 대부분 root 권한이 아니라서 Roxy가 반드시 필요하지는 않습니다. 다만 서비스를 관리할 일관된 방안이 필요합니다.
2. Piglet은 root 권한이 필요해서 실행할 때마다 패스워드를 묻기때문에 roxy 없이는 서비스 제어가 어렵습니다.
